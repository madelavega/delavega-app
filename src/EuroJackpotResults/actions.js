const actionPrefix = 'EURO_JACKPOT_RESULTS';
const genAction = (action) => `${actionPrefix}_${action}`;

export const actions = {
    LOAD_RESULTS: genAction('LOAD_RESULTS'),
};

export const actionCreators = {
    /**
     *
     * @param {String} date
     * @returns {{payload: {date}, type: string}}
     */
    loadResults: (date) => ({
        type   : actions.LOAD_RESULTS,
        payload: { date },
    }),
};

