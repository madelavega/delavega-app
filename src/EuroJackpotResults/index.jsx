import React, { useEffect, useRef } from 'react';
import PropTypes from 'prop-types';
import 'delavega-lib';

function JacksPotResults(props) {
    const table = useRef();

    /**
     *
     */
    useEffect(() => {
        const { data } = props,
            currentNode = table.current;
        currentNode.data = data;
    }, ['data']);

    return (
        <div>
            <delavega-table ref={table}>
                <delavega-table-column header="Tier" dataIndex="tier"/>
                <delavega-table-column header="Match" dataIndex="match"/>
                <delavega-table-column header="Winners" dataIndex="winners"/>
                <delavega-table-column header="Amount" dataIndex="amount"/>
            </delavega-table>
        </div>
    );
}

JacksPotResults.propTypes = {
    data: PropTypes.arrayOf(PropTypes.shape({
        tier   : PropTypes.string,
        amount : PropTypes.string,
        match  : PropTypes.string,
        winners: PropTypes.string,
    })),
};

export default JacksPotResults;