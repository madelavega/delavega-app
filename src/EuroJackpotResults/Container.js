import EuroJackpotResults from './index';
import { connect } from 'react-redux';
import { actionCreators } from '../DateSelector/actions';

const mapStateToProps = (state) => {
    const {
        euroJackpotResults : { data },
    } = state;

    return {
        data,
    };
};


const mapDispatchToProps = dispatch => {
    return {
        loadResults: (selectedDate) => dispatch(actionCreators.loadResults(selectedDate)),
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(EuroJackpotResults);
