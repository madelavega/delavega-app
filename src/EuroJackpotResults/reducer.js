import { actions } from './actions.js';

const initialState = {
    data: [
        { tier: 'I', match: '5 numbers + 2 euronumbers', winners: '0x', amount: '1000000' },
        { tier: 'II', match: '5 numbers + 1 euronumbers', winners: '3x', amount: '1000' },
        { tier: 'III', match: '5 numbers', winners: '3x', amount: '10000' },
        { tier: 'IV', match: '4 numbers + 1 euronumbers', winners: '13x', amount: '100000' },
    ],
};

const reducer = (state = initialState, action) => {
    const actionTypes = {
        [actions.LOAD_RESULTS]: () => {  // we won't reset on start, only on hide (using RESET_AND_HIDE)
            const { payload: { data } = {} } = action;
            return {
                ...state,
                data,
            };
        },
    };
    return actionTypes[action.type] ? actionTypes[action.type]() : { ...state };
};

export default reducer;
