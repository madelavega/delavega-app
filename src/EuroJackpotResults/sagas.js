import 'regenerator-runtime/runtime';
import { put, takeLatest } from 'redux-saga/effects';
import { actions, actionCreators } from './actions';
import Api from '../utils/Api';
import moment from 'moment';

/**
 * Do a request to appEnvConfig.endpoints.euroJackpotResults endpoinf (@see /config/{webpack_mode/services.js})
 * when the LOAD_RESULTS_REQUESTED action is dispatched
 * 
 * @param {object} action LOAD_RESULTS_REQUESTED @see actions.LOAD_RESULTS_REQUESTED
 * @returns {Generator<Promise<commander.ParseOptionsResult.unknown>|SimpleEffect<"PUT", PutEffectDescriptor<{payload: {data}, type: string}>>, void, *>}
 */
export const loadEuroJackpotResults = function* (action) {
    const { payload: { date } } = action;
    try {
        const { data } = (yield Api.get(appEnvConfig.endpoints.euroJackpotResults, {
            //maybe I could avoid the moment dependency, because the date format accepted
            //by the server is the reverse date from the combobox. So, date.split('-').reverse()
            //could be enough, but I prefer use moment tu allow i18n, for example
            urlParams: { date: moment(date, 'MM-DD-YYYY').format('YYYYDDMM') },
        }));
        yield put(actionCreators.loadResultsDone(data));
    } catch (ex) {
        yield put(actionCreators.loadResultsFailed( ex ));
    }
};

const euroJackpotSagas = [
    takeLatest(actions.LOAD_RESULTS_REQUESTED, loadEuroJackpotResults),
];

export default euroJackpotSagas;
