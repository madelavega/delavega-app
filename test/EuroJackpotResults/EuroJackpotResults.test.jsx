import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import EuroJackpotResults from '../../src/EuroJackpotResults';
const urlParams = [ '/results', '01-03-2021' ];

/**
 * Because EuroJackpotResults basically is a wrapper of a web component, which is managed with a reference,
 * we only test with jest and react-test-renderer that the EuroJackpotResults is rendered correctly,
 * without sending any prop that the referred web component uses. Its useless test it using nodeMocks (@see
 * https://reactjs.org/docs/test-renderer.html#ideas) because the real logic is in the web component. So,
 * it is tested in the library with all their options
 *
 */
describe('[EuroJackpotResult] renders correctly', () => {
    it('[EuroJackpotResult | Large] renders correctly', () => {
        const tree = renderer
            .create(
                <MemoryRouter initialEntries={urlParams}>
                    <EuroJackpotResults />
                </MemoryRouter>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});