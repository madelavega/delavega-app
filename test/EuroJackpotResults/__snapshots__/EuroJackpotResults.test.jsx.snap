// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[EuroJackpotResult] renders correctly [EuroJackpotResult | Large] renders correctly 1`] = `
<div>
  <span>
    Loading...
  </span>
</div>
`;
